restapi/datasource_api_object.go

```diff
@@ -115,34 +131,75 @@ func dataSourceRestApiRead(d *schema.ResourceData, meta interface{}) error {
     } /* End Loop through parts */
 
     data_array = (*ptr).([]interface{})
-  } else {
+  } else if "" == results_key && !single_obj {
     data_array = result.([]interface{})
+  } else { // single_obj
+    ptr := &result
+    single_object_data = (*ptr).(map[string]interface{})
   }
 
-  /* Loop through all of the results seeking the specific record */
-  for _, item := range data_array {
-    hash := item.(map[string]interface{})
-
-    /* We found our record */
-    if hash[search_key] == search_val {
-      id = hash[id_attribute].(string)
+  if !single_obj {
+
+    /* Loop through all of the results seeking the specific record */
+    for _, item := range data_array {
+      hash := item.(map[string]interface{})
+
+      // Parse search_key
+      search_parts := strings.Split(search_key, "/")
+      search_part := ""
+      search_seen := ""
+      search_hash := hash
+
+      // Loop through search parts
+      for len(search_parts) > 1 {
+        log.Printf("datasource_api_object.go: Looping through search_parts\n")
+        search_part, search_parts = search_parts[0], search_parts[1:]
+
+        // Protect against double slashes by mistake
+        if "" == search_part { break }
```

Comment:
Note that this doesn't protect against double-slashes so much as causes an end to processing if one is encountered in the middle of the string. This may or may not be desired.

Reply:
I was imitating what you were doing for the parsing of results_key.   You had:

```
      /* Protect against double slashes by mistake */
      if "" == part { break }
```
Perhaps it would be better to use "continue" in both cases, causing execution to go to the next iteration of the loop instead of stopping the for loop? For both loops, having an empty string for part or search_part will cause problems, so we definitely need to at least do "continue".

On the other hand, users should not be putting in double slashes.  Was your original concern that they might set results_key from the output of some other resource which could include double slashes?

Reply:
Yeah, a break here, and later, is definitely the safe maneuver. I was thinking more about just being forgiving of what users may put in. I seem to recall that the initial logic used `continue`, but I switched to `break` to get out of the loop once the end of the path was reached... basically defeating my goal of tolerating double slashes (oops).

We can stick with break for now. If we decide to be more tolerant of bad input later, we could adjust this to say something along the lines of "if part is empty but is not the last item in the array, then continue"